_posts/2022-02-10-Grant-Admin-Rights.md

```diff
@@ -0,0 +1,153 @@
+---
+layout: post
+title: Grant Administrator Right to User
+description: Grant sudo/administrator right to user and its bound systems
+tags:
+  - powershell
+  - settings
+  - automation
+---
+The purpose of this script is to grant admin access to a JumpCloud user to a bound system.
+This script prompts the user to enter username, system Id, and minutes of how long the user is going to be an admin/sudo. After granted admin status, the user will need to re-login if on Windows machine to see the admin access change. The user is revoked admin status after x mins. User must re-login if on Windows machine to be demoted to standard user. 
+
+### Basic Usage
+
+* The PowerShell Module is required to run this script
+* Save the script to your desired location
+* In a PowerShell terminal window, run the script `~/Path/To/Scrpt/filename.ps1`
+* Follow the prompts to enter:
+    * Username
+    * System Id
+    * Amount of minutes user is granted admin access
+* User will need to re-login if on Windows Machine to be granted/revoked admin access
+
+### Additional Information
+
+* Run the script from your desired directory
+* Mac and Linux machines will not need to re-login
+
+* Powershell prompts to grant admin rights
+![Grant admin prompts](./../../../images/grantAdminPrompts.gif)
+
+
+* JumpCloud portal permission change
+![Grant admin Jc portal](./../../../images/grantAdminJcPortal.gif)
+### Script
+
+```powershell
+function Get-User 
+{
+    $Username = Read-Host "Please enter the target JumpCloud User you'd like to elevate to admin/sudo temporarily"
+
+    # Get users
+    $users = Get-JCUser -UserName $UserName
```

Comment:
The other nice benefit of using Get-JCUser here is that it calls the Connect-JCOnline function and prompts the user to enter their API key
<img width="1143" alt="Screen Shot 2022-02-14 at 3 28 46 PM" src="https://user-images.githubusercontent.com/54448601/153957489-3fd06171-7e7f-4218-aff7-60753036cbc4.png">